packages/grafana-toolkit/src/plugins/manifest.ts

```diff
@@ -0,0 +1,87 @@
+import path from 'path';
+import fs from 'fs';
+import crypto from 'crypto';
+import { ManifestInfo } from './types';
+
+const MANIFEST_FILE = 'MANIFEST.txt';
+
+async function* walk(dir: string, baseDir: string): Promise<string> {
+  for await (const d of await (fs.promises as any).opendir(dir)) {
+    const entry = path.join(dir, d.name);
+    if (d.isDirectory()) {
+      yield* await walk(entry, baseDir);
+    } else if (d.isFile()) {
+      yield path.relative(baseDir, entry);
+    } else if (d.isSymbolicLink()) {
+      const realPath = fs.realpathSync(entry);
+      if (!realPath.startsWith(baseDir)) {
+        throw new Error(
+          `symbolic link ${path.relative(baseDir, entry)} targets a file outside of the base directory: ${baseDir}`
+        );
+      }
+      yield path.relative(baseDir, entry);
+    }
+  }
+}
+
+export async function buildManifest(dir: string): Promise<ManifestInfo> {
+  const pluginJson = JSON.parse(fs.readFileSync(path.join(dir, 'plugin.json'), { encoding: 'utf8' }));
+
+  const manifest = {
+    plugin: pluginJson.id,
+    version: pluginJson.info.version,
+    files: {},
+  } as ManifestInfo;
+
+  for await (const p of walk(dir, dir)) {
+    if (p === MANIFEST_FILE) {
+      continue;
+    }
+
+    if (/\.(png|gif|svg)$/.test(p)) {
```

Comment:
FYI: this filter was a bit add hoc.  I think we should avoid signing large files that do not add any real security.

Reply:
Hashing the files is pretty fast since it's just a single sha256sum (rather than a deliberately-expensive password hashing algorithm), so I'd like to err on the side of protecting as many files as possible.  This list of exclusions was just taken for the current code, but there is an argument to be made that we should just hash all files in the plugin.

Reply:
If you dont hash all of the files in the plugin folder, Grafana wont be able to verify that the plugin folder only contains files that should be present.  Untrusted files in the plugin folder are still a security issue, as a malicious user could link directly to them, allowing them to trick a user into running malicious code in the browser.

So, we should hash all files.

Reply:
+1 to change it to all files now.  We can have signatures with type 2.0+ require all files.

Reply:
works4me, will update

Reply:
we may also want to look at what we do with grafana itself as it has all the same issues!

Reply:
yeah I think that can be a 2-stage approach, for v2+ signatures we can require that all files are signed, while leaving the v1 behavior alone, then at some point we remove support for v1 signatures.